File: source/clibCore/addonCollection_core/blockEmail.vb

Option Explicit On
Option Strict On

Imports Contensive.BaseClasses
Imports Contensive.Core
Imports Contensive.Core.Models.Entity
Imports Contensive.Core.Controllers
Imports Contensive.Core.Controllers.genericController
'
Namespace Contensive.Addons.Core
    Public Class blockEmailClass
        Inherits Contensive.BaseClasses.AddonBaseClass
        '
        '====================================================================================================
        ''' <summary>
        ''' getFieldEditorPreference remote method
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <returns></returns>
        Public Overrides Function execute(cp As Contensive.BaseClasses.CPBaseClass) As Object
            Dim result As String = ""
            Try
                Dim processor As CPClass = DirectCast(cp, CPClass)
                Dim cpCore As coreClass = processor.core
                '
                ' -- click spam block detected
                If True Then
                    '
                    Dim recipientEmailToBlock As String = cpCore.docProperties.getText(rnEmailBlockRecipientEmail)
                    If String.IsNullOrEmpty(recipientEmailToBlock) Then
                        Dim recipientList As List(Of personModel) = personModel.createList(cpCore, "(email=" & cpCore.db.encodeSQLText(recipientEmailToBlock) & ")")
                        For Each recipient In recipientList
                            recipient.AllowBulkEmail = False
                            recipient.save(cpCore)
                            '
                            ' -- Email spam footer was clicked, clear the AllowBulkEmail field
                            Call cpCore.email.addToBlockList(recipientEmailToBlock)
                            '
                            ' -- log entry to track the result of this email drop
                            Dim emailDropId As Integer = cpCore.docProperties.getInteger(rnEmailBlockRequestDropID)
                            If emailDropId <> 0 Then
                                Dim emailDrop As emailDropModel = emailDropModel.create(cpCore, emailDropId)
                                If (emailDrop IsNot Nothing) Then
                                    Dim log As New Models.Entity.emailLogModel() With {
                                        .name = "User " & recipient.name & " clicked linked spam block from email drop " & emailDrop.name & " at " & cpCore.doc.profileStartTime.ToString(),
                                        .EmailDropID = emailDrop.id,
                                        .MemberID = recipient.id,
                                        .LogType = EmailLogTypeBlockRequest
                                    }
                                End If
                            End If
                            Call cpCore.webServer.redirect(cpCore.webServer.requestProtocol & cpCore.webServer.requestDomain & "/ccLib/popup/EmailBlocked.htm", "Group Email Spam Block hit. Redirecting to EmailBlocked page.", False)
                        Next
                    End If
                End If
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace

File: source/clibCoreCs/addonCollection_core/blockEmail.cs
using System;

//========================================================================
// This conversion was produced by the Free Edition of
// Instant C# courtesy of Tangible Software Solutions.
// Order the Premium Edition at https://www.tangiblesoftwaresolutions.com
//========================================================================

using Contensive.BaseClasses;
using Contensive.Core;
using Models.Entity;
using Contensive.Core.Controllers;

//
namespace Contensive.Addons.Core
{
	public class blockEmailClass : Contensive.BaseClasses.AddonBaseClass
	{
		//
		//====================================================================================================
		/// <summary>
		/// getFieldEditorPreference remote method
		/// </summary>
		/// <param name="cp"></param>
		/// <returns></returns>
		public override object Execute(Contensive.BaseClasses.CPBaseClass cp)
		{
			string result = "";
			try
			{
				CPClass processor = (CPClass)cp;
				coreClass cpCore = processor.core;
				//
				// -- click spam block detected
				if (true)
				{
					//
					string recipientEmailToBlock = cpCore.docProperties.getText(rnEmailBlockRecipientEmail);
					if (string.IsNullOrEmpty(recipientEmailToBlock))
					{
						List<personModel> recipientList = personModel.createList(cpCore, "(email=" + cpCore.db.encodeSQLText(recipientEmailToBlock) + ")");
						foreach (var recipient in recipientList)
						{
							recipient.AllowBulkEmail = false;
							recipient.save(cpCore);
							//
							// -- Email spam footer was clicked, clear the AllowBulkEmail field
							cpCore.email.addToBlockList(recipientEmailToBlock);
							//
							// -- log entry to track the result of this email drop
							int emailDropId = cpCore.docProperties.getInteger(rnEmailBlockRequestDropID);
							if (emailDropId != 0)
							{
								emailDropModel emailDrop = emailDropModel.create(cpCore, emailDropId);
								if (emailDrop != null)
								{
									Models.Entity.emailLogModel log = new Models.Entity.emailLogModel()
									{
										name = "User " + recipient.name + " clicked linked spam block from email drop " + emailDrop.name + " at " + cpCore.doc.profileStartTime.ToString(),
										EmailDropID = emailDrop.id,
										MemberID = recipient.id,
										LogType = EmailLogTypeBlockRequest
									};
								}
							}
							return cpCore.webServer.redirect(cpCore.webServer.requestProtocol + cpCore.webServer.requestDomain + "/ccLib/popup/EmailBlocked.htm", "Group Email Spam Block hit. Redirecting to EmailBlocked page.");
						}
					}
				}
			}
			catch (Exception ex)
			{
				cp.Site.ErrorReport(ex);
			}
			return result;
		}
	}
}
